helm/a3s/templates/deployment.yaml

```diff
@@ -0,0 +1,164 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: a3s
+spec:
+  replicas: {{ .Values.replicas | default 1}}
+  strategy:
+    type: {{ required "updateStrategy is required" .Values.updateStrategy }}
+  selector:
+    matchLabels:
+      app: a3s
+      type: core
+  template:
+    metadata:
+      labels:
+        app: a3s
+        type: core
+    spec:
+      terminationGracePeriodSeconds: {{ required "global.terminationGracePeriodSeconds is required" .Values.global.terminationGracePeriodSeconds }}
+      containers:
+        - name: a3s
+          image: {{ required "global.imageRegistry is required" .Values.global.imageRegistry }}/a3s:{{ required "global.imageTag is required" .Values.global.imageTag }}
+          imagePullPolicy: "Always"
+          env:
+            # General
+            - name: A3S_LISTEN
+              value: ":1443"
+            - name: A3S_ENCODING
+              value: "msgpack"
```

Comment:
this should be configurabnle. also since msgpack is default, we should rely on flags defaults, not charts defaults